needle/modules/storage/data/files_mdm.py

```diff
@@ -0,0 +1,128 @@
+        out = self.device.remote_op.command_blocking(cmd)
+        retrieved_files = out
+
+        return retrieved_files
+
+    # Filter out duplicate directories, and return list
+    def filter_dirs(self, dirs):
+        directories = []; 
+        [directories.append(Utils.get_file_path(x)) for x in dirs if Utils.get_file_path(x) not in directories]
+        if not self.options['bulk']: directories.append('All')
+
+        return directories
+
+    # ==================================================================================================================
+    # RUN
+    # ==================================================================================================================
+    def module_run(self):
+        self.printer.info("Looking for MDM Configuration file locations...")
+
+        # Find MDM config file locations
+        cmd = '{bin} / -type f -name "MDM*.plist" -o -name "UserSettings.plist"'.format(bin=self.device.DEVICE_TOOLS['FIND'])
+        out = self.device.remote_op.command_blocking(cmd)
+
+        if not out:
+            self.printer.info("No Configuration files found")
+            return
+
+        directories = self.filter_dirs(out)
+
+        if not self.options['bulk']:
+            self.individ_mode(directories)
+        else:
+            self.printer.warning("Running in bulk mode!")
+            self.bulk_mode(directories)
+
+    # Run standard mode
+    def individ_mode(self, directories):
+        dirs_str = self.select_dir(directories)
+        retrieved_files = self.get_files(dirs_str)
+
+        # Begin "individual file" user interactivity loop
+        while True and not self.options['bulk']:
+            option = choose_from_list(retrieved_files) 
+
+            # Run plutil on selected file
+            self.printer.info("Dumping content of the file...")
+            pl = dict(self.device.remote_op.parse_plist(option))
+
+            # Initialize output filename
+            outfile = ""
+            if self.options['output']:
+                fileName = Utils.extract_filename_from_path(option.strip())
+                outfile = str(self.options['output']+"/"+fileName.replace('.', '_export.'))
+            else: 
+                outfile = None
+
+            # Print Data to user
+            self.print_cmd_output(pl)
+
+            if self.options['autosave']:
+                self.device.pull(option.strip(), outfile)
+            else:
+                save = choose_yes_no("Would you like to %spull%s ths file?" % (Colors.R, Colors.N))
+                if save: self.device.pull(option.strip(), outfile)
+                    
+            # Ask user if they wish to view another file
+            yes = choose_yes_no("Would you like to inspect another file?")
+            if not yes: break
+
+    # Run bulk gather mode
+    def bulk_mode(self, directories):
+        while True:
+            dirs_str = self.select_dir(directories)
```

Comment:
In addition, great part of this function share code from the previous one. At least part of them can be merged